### PR TITLE
[Alerting]: namespace fixes

### PR DESCRIPTION
**Pull request by papagian**
**What this PR does / why we need it**:
- forbid viewers for updating rules if viewers can edit (`viewers_can_edit` is set)
- clear ngalert tables when deleting the folder
- add tests

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/tests/api/alerting/api_alertmanager_test.go [hunk 1]
```diff
@@ -464,6 +464,256 @@ func TestAlertAndGroupsQuery(t *testing.T) {
...
+	// Create the namespace we'll save our alerts to.
+	namespaceUID, err := createFolder(t, store, 0, "default")
+	require.NoError(t, err)
+
+	require.NoError(t, createUser(t, store, models.ROLE_VIEWER, "viewer", "viewer"))
+	require.NoError(t, createUser(t, store, models.ROLE_EDITOR, "editor", "editor"))
+
+	createRule(t, grafanaListedAddr, "default", "editor", "editor")
+
+	// check rule is created
```
Comment (gotjosh) [inline pkg/tests/api/alerting/api_alertmanager_test.go:606]:
```suggestion
	// First, let's have an editor create a rule within the folder/namespace.
```

File: pkg/tests/api/alerting/api_alertmanager_test.go [hunk 2]
```diff
@@ -464,6 +464,256 @@ func TestAlertAndGroupsQuery(t *testing.T) {
...
+							}
+						}
+					]
+				}
+			]
+		}`, namespaceUID)
+		assert.JSONEq(t, expectedGetRulesResponseBody, string(b))
+	}
+
+	// delete folder
```
Comment (gotjosh) [inline pkg/tests/api/alerting/api_alertmanager_test.go:681]:
```suggestion
	// Next, the editor can delete the folder.
```

File: pkg/tests/api/alerting/api_alertmanager_test.go [hunk 3]
```diff
@@ -464,6 +464,256 @@ func TestAlertAndGroupsQuery(t *testing.T) {
...
+			err := resp.Body.Close()
+			require.NoError(t, err)
+		})
+		b, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Equal(t, 200, resp.StatusCode)
+		require.JSONEq(t, `{"id":1,"message":"Folder default deleted","title":"default"}`, string(b))
+	}
+
+	// check alert rule is deleted
```
Comment (gotjosh) [inline pkg/tests/api/alerting/api_alertmanager_test.go:699]:
```suggestion
	// Finally, we ensure the rules were deleted.
```

**Review by gotjosh — APPROVED** [review]
LGTM

File: pkg/services/ngalert/store/alert_rule.go [hunk 4]
```diff
@@ -368,16 +368,16 @@ func (st DBstore) GetRuleGroupAlertRules(query *ngmodels.ListRuleGroupAlertRules
...
 	folder, err := s.GetFolderByTitle(namespace)
 	if err != nil {
 		return nil, err
 	}
 
-	if withEdit {
+	if withCanSave {
 		g := guardian.New(folder.Id, orgID, user)
-		if canAdmin, err := g.CanEdit(); err != nil || !canAdmin {
+		if canSave, err := g.CanSave(); err != nil || !canSave {
```
Comment (marefr) [inline pkg/services/ngalert/store/alert_rule.go:380]: Nit. Think at least logging the returned error would be good in case it is of importance.

**Review by marefr — APPROVED** [review]
LGTM added a nit comment to consider

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**